src/renderers/action-parse.renderer.ts

```diff
@@ -0,0 +1,107 @@
+import domino from 'domino'
+import { DownloadOpts, DownloadRes, Renderer } from './abstract.renderer.js'
+import { RenderOpts, RenderOutput } from './abstract.renderer.js'
+import { config } from '../config.js'
+import { genCanonicalLink, genHeaderScript, genHeaderCSSLink, getStaticFiles } from '../util/misc.js'
+import MediaWiki from '../MediaWiki.js'
+import { htmlVectorLegacyTemplateCode, htmlVector2022TemplateCode } from '../Templates.js'
+import Downloader from '../Downloader.js'
+
+// Represent 'https://{wikimedia-wiki}/w/api.php?action=parse&format=json&prop=modules|jsconfigvars|text&parsoid=1&page={article_title}&skin=vector-2022'
+export class ActionParseRenderer extends Renderer {
+  public staticFilesList: string[] = []
+  constructor() {
+    super()
+    if (this.staticFilesList.length === 0) {
+      this.staticFilesList = getStaticFiles(config.output.jsResourcesCommon, config.output.cssResourcesCommon)
+    }
+  }
+
+  public templateDesktopArticle(moduleDependencies: any, articleId: string): Document {
+    const { jsConfigVars, jsDependenciesList, styleDependenciesList } = moduleDependencies as {
+      jsConfigVars
+      jsDependenciesList: string[]
+      styleDependenciesList: string[]
+    }
+
+    const htmlTemplateCode = MediaWiki.skin === 'vector' ? htmlVectorLegacyTemplateCode : MediaWiki.skin === 'vector-2022' ? htmlVector2022TemplateCode : null
+
+    if (!htmlTemplateCode) {
+      throw new Error(`Skin ${MediaWiki.skin} is not supported by ActionParse renderer`)
+    }
+    const htmlTemplateString = htmlTemplateCode()
+      .replace('__CSS_LINKS__', '') // Unused with ActionParse
```

Comment:
If they're not used, why are they in the templates that you're adding in this PR. Just leave them out, right?

Reply:
Done